### PR TITLE
Add definitions of pre-releases

### PR DESCRIPTION
**Pull request by Oshratn**
Towards #21081

Added words to describe pre-releases. Towards creating a single source of truth for this information and deprecating the related intranet page: https://intranet.giantswarm.io/docs/product/releases/versioning-scope-lifecycle/

### Please remember to

- [ ] Give the PR a meaningful title -- it will be shown in the release notes
- [ ] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/main/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

File: src/content/general/releases/index.md [hunk 1]
```diff
@@ -87,13 +91,29 @@ Once we provide a new workload cluster release, be it major, minor, or patch, we
...
+
+Currently not all Giant Swarm releases become available as pre-releases. We communicate pre-releases, when they are available for customer testing.
+
+Pre-releases can be alpha, beta, or release candidates (RC). This depends on the needs of the team wanting to test their feature.
+
+Use Cases for Pre-Releases:
+
+- A new feature with unclear impact is done and the team wants to test and evaluate its impact before deciding the scope of the release (alpha). Customers get early access to the functionaliy and see if it satisfies their requirements. Customers report back with bugs and additional commentary. This allows the functionality to be evaluated in a less controlled way than in our testing environments. It also identifies bugs that we may have not found, allowing us to fix them early. It also may bring about reconsidering the feature/solution in general.
```
Comment (alex-dabija) [inline src/content/general/releases/index.md:104]:
```suggestion
- A new feature with unclear impact is done and the team wants to test and evaluate its impact before deciding the scope of the release (alpha). Customers get early access to the functionality and see if it satisfies their requirements. Customers report back with bugs and additional commentary. This allows the functionality to be evaluated in a less controlled way than in our testing environments. It also identifies bugs that we may have not found, allowing us to fix them early. It also may bring about reconsidering the feature/solution in general.
```

File: src/content/general/releases/index.md [hunk 2]
```diff
@@ -87,13 +91,29 @@ Once we provide a new workload cluster release, be it major, minor, or patch, we
 
 Once deprecated, you can still continue to use the workload cluster release with existing clusters. In addition, as long as you have at least one cluster in your installation running the deprecated release, you can also create new clusters using that release. This allows you to create test clusters to test an upgrade from that release.
 
+### Pre-releases
+
+Pre-releases (when available) give you the option to test new features or use new functionality before the feature is released in a major version. At the same time helps the us release new features and test them in a realworld scenario with your help. When you test out a pre-release it helps find bugs or evaluate the impact of new functionality on the system.
```
Comment (alex-dabija) [inline src/content/general/releases/index.md:96]: We did do preview releases for only major ones until now but I wouldn't limit them just to majors. I can think of doing a minor preview release for an optional feature.

**Review by alex-dabija — APPROVED** [review]
It looks good to me, beside the 2 comments I made.

File: src/content/general/releases/index.md (hunk 2 above)
Comment (Oshratn) [reply src/content/general/releases/index.md]: Removed the limitation to major releases.

**Review by alex-dabija — APPROVED** [review]

**Comment by Oshratn** [comment]
Thanks Alex.
@marians can you take a look at the inspecting section, so i can merge an up to date document. Thanks in advance 💐

**Review by marians — APPROVED** [review]
LGTM

**Comment by Oshratn** [comment]
@marians I am circling back on this to ask if you only looked at the diff or at the full document?
In the full document there is a 🌈 related section which seemed out of date and I wanted to kill 2 birds with one PR. Did you have a look at that section?

**Pull request merged**